test/parallel_api/algorithm/alg.nonmodifying/transform_if.pass.cpp

```diff
@@ -0,0 +1,200 @@
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, mask, result, mutable_negate_first<value_type1, value_type2>{},
+                                  mutable_check_mask_second<value_type1, value_type2>{});
+
+        EXPECT_TRUE(check(first, last, mask, result, n), "transform_if binary wrong result");
+        // reset output elements to 0
+        ::std::fill(result, result_end, 0);
+    }
+};
+
+struct test_transform_if_unary
+{
+    template <typename It1, typename Out, typename Size>
+    bool
+    check(It1 first, It1 last, Out result, Size n)
+    {
+
+        for (; first != last; ++first, ++result)
+        {
+            if (*first % 3 == 0 && *result != -(*first))
+            {
+                return false;
+            }
+            else if (*first % 3 != 0 && *result != 0)
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    template <typename Policy, typename InputIterator1, typename OutputIterator, typename Size>
+    void
+    operator()(Policy&& exec, InputIterator1 first, InputIterator1 last, OutputIterator result,
+               OutputIterator result_end, Size n)
+    {
+        using value_type1 = typename ::std::iterator_traits<InputIterator1>::value_type;
+        // call transform_if
+        oneapi::dpl::transform_if(exec, first, last, result, mutable_negate<value_type1>{},
+                                  mutable_check_mod3_is_0<value_type1>{});
+
+        EXPECT_TRUE(check(first, last, result, n), "transform_if unary wrong result");
+        // reset output elements to 0
+        ::std::fill(result, result_end, 0);
+    }
+};
+
+template <typename In1, typename In2, typename Out>
+void
+test()
+{
+    for (size_t n = 1; n <= 100000; n = n <= 16 ? n + 1 : size_t(3.1415 * n))
+    {
+        {
+            Sequence<In1> in1(n, [=](size_t k) { return (3 * k); });
+            Sequence<In2> in2(n, [=](size_t k) { return k % 2 == 0 ? 1 : 0; });
+
+            Sequence<Out> out(n, [=](size_t) { return 0; });
+
+            invoke_on_all_policies<0>()(test_transform_if_binary(), in1.begin(), in1.end(), in2.begin(), in2.end(),
+                                        out.begin(), out.end(), n);
+            invoke_on_all_policies<1>()(test_transform_if_binary(), in1.cbegin(), in1.cend(), in2.cbegin(), in2.cend(),
+                                        out.begin(), out.end(), n);
+        }
+        {
+            Sequence<In1> in1(n, [=](size_t k) { return k; });
+            Sequence<Out> out(n, [=](size_t) { return 0; });
+
+            invoke_on_all_policies<0>()(test_transform_if_unary(), in1.begin(), in1.end(), out.begin(), out.end(), n);
+            invoke_on_all_policies<1>()(test_transform_if_unary(), in1.cbegin(), in1.cend(), out.begin(), out.end(), n);
+        }
+    }
+}
+
+int
+main()
+{
+    test<::std::int32_t, ::std::int32_t, ::std::int32_t>();
```

Comment:
What if ```TEST_DPCPP_BACKEND_PRESENT``` is undefined?

Reply:
If `TEST_DPCPP_BACKEND_PRESENT` is undefined, I believe `invoke_on_all_policies` will just run the tests on `invoke_on_all_host_policies` and skip the `invoke_on_all_hetero_policies` call.  Am I missing some other required guard?